src/spyglass/lfp/lfp_imported.py

```diff
@@ -15,14 +26,105 @@ class ImportedLFP(SpyglassMixin, dj.Imported):
     """
 
+    _nwb_table = Nwbfile
+
     def make(self, key):
         """Placeholder for importing LFP."""
         raise NotImplementedError(
             "For `insert`, use `allow_direct_insert=True`"
         )
+
+    def _no_transaction_make(self, key):
+
+        nwb_file_name = key["nwb_file_name"]
+        nwb_file_abspath = Nwbfile.get_abs_path(nwb_file_name)
+        nwbf = get_nwb_file(nwb_file_abspath)
+
+        # get the set of lfp objects in the file
+        lfp_objects = []
+        for obj in nwbf.objects:
+            if isinstance(obj, pynwb.ecephys.LFP):
+                lfp_objects.append(obj)
+
+        if len(lfp_objects) == 0:
+            logger.warning(
+                f"No LFP objects found in {nwb_file_name}. Skipping."
+            )
+            return
+        # get the electrical series objects from the lfp objects
+        lfp_es_objects = []
+        for lfp_object in lfp_objects:
+            lfp_es_objects.extend(list(lfp_object.electrical_series.values()))
+
+        for i, es_object in enumerate(lfp_es_objects):
+            electrodes_df = es_object.electrodes.to_dataframe()
+            electrode_ids = electrodes_df.index.values
+
+            # check if existing electrode group for this set of electrodes exists
+            e_group_query = LFPElectrodeGroup() & {
+                "nwb_file_name": nwb_file_name,
+            }
+            group_key = None
+            for test_group_key in e_group_query.fetch("KEY"):
+                group_electodes = (
+                    LFPElectrodeGroup().LFPElectrode() & test_group_key
+                ).fetch("electrode_id")
+
+                if set(group_electodes) == set(electrode_ids):
+                    group_key = test_group_key
+                    break
+
+            if group_key is None:
+                # need to create a new group
+                group_num = len(
+                    e_group_query
+                    & "lfp_electrode_group_name LIKE 'imported_lfp_%'"
+                )
+                group_name = f"imported_lfp_{group_num:03}"
+                group_key = {
+                    "nwb_file_name": nwb_file_name,
+                    "lfp_electrode_group_name": group_name,
+                }
+                if len(LFPElectrodeGroup() & group_key):
+                    raise ValueError(
+                        f"LFPElectrodeGroup {group_key} already exists with different electrode entries."
+                    )
+
+                LFPElectrodeGroup().create_lfp_electrode_group(
+                    nwb_file_name=nwb_file_name,
+                    group_name=group_name,
+                    electrode_ids=electrode_ids,
+                )
+
+            # estimate the sampling rate or read in if available
+            sampling_rate = (
+                estimate_sampling_rate(es_object.timestamps[: int(1e6)])
+                if es_object.rate is None
+                else es_object.rate
+            )
```

Comment:
I find this more readable when providing a backup val for something that may be none

```suggestion
            sampling_rate = (
                es_object.rate or 
                estimate_sampling_rate(es_object.timestamps[: int(1e6)])
            )
```